themes/default/content/registry/packages/azuread/installation-configuration.md

```diff
@@ -59,9 +63,57 @@ Pick out the `<id>` from the list and run:
 $ az account set --subscription=<id>
 ```
 
-#### Authenticate using a Service Principal
 
-A Service Principal is an application in Azure Active Directory with three authorization tokens: a client ID, a client secret, and a tenant ID. Using a Service Principal is the recommended way to connect Pulumi to Azure in a team or CI setting.
+### Authenticate with OpenID Connect (OIDC)
+
+OIDC allows you to establish a trust relationship between Azure and another identity provider such as GitHub or Azure DevOps. Once
+established, your program can exchange an ID token issued by the identity provider for an Azure token. Your Pulumi program running in
+the identity provider's service, for instance, GitHub Actions CI or Azure DevOps Pipelines, can then access Azure, without storing any
+secrets in GitHub.
+
+#### OIDC Azure Configuration
+
+To configure the trust relationship in Azure, please refer to
+[this guide](https://learn.microsoft.com/en-us/azure/active-directory/workload-identities/workload-identity-federation-create-trust?pivots=identity-wif-apps-methods-azp#github-actions).
+This needs to be set up only once.
+
+#### OIDC Pulumi Provider Configuration
+
+To use OIDC, either set the Pulumi configuration `useOidc` via `pulumi config set azuread:useOidc true` or set the
+environment variable `ARM_USE_OIDC` to "true".
+
+Next, supply the provider with the ID token to exchange for an Azure token. There are three ways to do this depending on
+the service your program will run on.
+
+- In GitHub, we don't need to configure anything since
+[GitHub sets the relevant environment variables](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect)
+`ACTIONS_ID_TOKEN_REQUEST_TOKEN` and `ACTIONS_ID_TOKEN_REQUEST_URL` by default and the provider reads them automatically. 
+
+- Other identity providers offer a way to access the ID token. For instance, in GitLab CI/CD jobs, the ID token is available
+via the environment variable `GITLAB_OIDC_TOKEN`. Configure the Pulumi provider to use this token by setting the Pulumi
+configuration `azuread:oidcToken` or the environment variable `ARM_OIDC_TOKEN`.
+
+- In case your provider does not offer an ID token directly but instead a way to exchange a local bearer token for an ID
```

Comment:
For this section:

- In case your provider does not offer an ID token directly but instead a way to exchange a local bearer token for an ID token, you can configure this as well via the pair of `azuread:oidcRequestToken` or environment variable
`ARM_OIDC_REQUEST_TOKEN` and `azuread:oidcRequestUrl` or environment variable `ARM_OIDC_REQUEST_TOKEN`

I would recommend updating the verbiage a bit so that there is a bit more clarity around the pairing and that both are needed. maybe something like the following:

- If your provider does not offer an ID token directly but it does offer a way to exchange a local bearer token for an ID
token, you can configure the retrieval of the ID token by setting one of the following pairs:
    - both the `azuread:oidcRequestToken` and `azuread:oidcRequestUrl` Pulumi configuration values - OR -
    - both the `ARM_OIDC_REQUEST_TOKEN` and `ARM_OIDC_REQUEST_TOKEN` environment variables